### DISCUSSION TITLE
XML monitor client for Mac OSX

### DISCUSSION THREAD

**Discussion by testaccount00075433** [processone/ejabberd#3810] [Questions] [answered]
Hello guys,

1. Is there any client with XML monitor to see what santza was sent? 

2. Is it possible to set custom properties to chat room? For example custom id/type?

I need to have in my app one general module for all chat rooms and module where I have specified chat rooms for that custom id. How I can achieve that?

**Comment by badlop** [comment, marked as answer]
> Is there any client with XML monitor to see what santza was sent?

Many Jabber/XMPP clients include some kind of "XML console", Gajim, Psi, ... most of them.

When using other clients that don't include it, I've used tcpdump (simple and console), and Wireshark (graphical, many options).
You can search for other alternatives in this page: https://alternativeto.net/software/wireshark/

> Is it possible to set custom properties to chat room? For example custom id/type?

Looking at mod_muc options, it isn't possible to add more fields to the room configuration. But maybe you can use the "Description" field, and add there whatever details you want.

BTW, in ejabberd 21.12 the API change_room_option cannot change the 'description' option yet.
